applications/GeoMechanicsApplication/custom_conditions/T_microclimate_flux_condition.cpp

```diff
@@ -0,0 +1,448 @@
+{
+    Condition::Initialize(rCurrentProcessInfo);
+    InitializeProperties();
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TMicroClimateFluxCondition<TDim, TNumNodes>::InitializeSolutionStep(const ProcessInfo& rCurrentProcessInfo)
+{
+    // Unfortunately, the roughness temperature and net radiation cannot be
+    // initialized in the Initialize() method because the values of the air
+    // temperature and solar radiation are not yet available.
+    if (!mIsInitialized)
+    {
+        const GeometryType& Geom = this->GetGeometry();
+        mRoughnessTemperature = Geom[0].FastGetSolutionStepValue(AIR_TEMPERATURE, 0);
+        mNetRadiation = Geom[0].FastGetSolutionStepValue(SOLAR_RADIATION, 0);
+        mIsInitialized = true;
+    }
+
+    CalculateRoughness(rCurrentProcessInfo);
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TMicroClimateFluxCondition<TDim, TNumNodes>::CalculateLocalSystem(
+    Matrix& rLeftHandSideMatrix, Vector& rRightHandSideVector, const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    rLeftHandSideMatrix = Matrix{TNumNodes, TNumNodes, 0.0};
+    rRightHandSideVector = Vector{TNumNodes, 0.0};
+
+    // Previous definitions
+    const auto& r_geom = this->GetGeometry();
+    const auto& r_integration_points =
+        r_geom.IntegrationPoints(this->GetIntegrationMethod());
+    const auto number_of_integration_points =
+        static_cast<unsigned int>(r_integration_points.size());
+
+    // Containers of variables at all integration points
+    const auto local_dim = static_cast<unsigned int>(r_geom.LocalSpaceDimension());
+    GeometryType::JacobiansType J_container(number_of_integration_points);
+    for (unsigned int i = 0; i < number_of_integration_points; ++i)
+        J_container[i].resize(TDim, local_dim, false);
+    r_geom.Jacobian(J_container, this->GetIntegrationMethod());
+
+    const auto& r_N_container =
+        this->GetGeometry().ShapeFunctionsValues(this->GetIntegrationMethod());
+
+    auto nodal_temperatures = array_1d<double, TNumNodes>{};
+    VariablesUtilities::GetNodalValues(this->GetGeometry(), TEMPERATURE,
+                                       nodal_temperatures.begin());
+
+    const auto time_step_size = rCurrentProcessInfo.GetValue(DELTA_TIME);
+
+    const auto previous_storage = mWaterStorage;
+    const auto previous_radiation = mNetRadiation;
+    mWaterStorage = CalculateCurrentWaterStorage(
+        time_step_size, previous_storage, previous_radiation);
+    mNetRadiation = CalculateCurrentNetRadiation();
+
+    const auto left_hand_side_fluxes = CalculateLeftHandSideFluxes();
+    const auto right_hand_side_fluxes = CalculateRightHandSideFluxes(
+        time_step_size, previous_storage, previous_radiation);
+
+    // Loop over integration points
+    for (unsigned int integration_point_index = 0;
+         integration_point_index < number_of_integration_points; ++integration_point_index)
+    {
+        const auto N =
```

Comment:
Maybe we can think of a better name for N?

Reply:
In finite elements N is the general notation for the interpolation polynomial, it matches with almost all textbooks. Therefore please let N remain N.

Reply:
Okay, clear we'll keep it like it is